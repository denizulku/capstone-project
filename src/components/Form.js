import React, { useState } from 'react'
import styled from 'styled-components'
import List from './List'

export default function Form() {
  const [habits, setHabits] = useState([])
  return (
    <>
      <Container onSubmit={handleSubmit}>
        <Input name="habit" type="text" placeholder="Insert a habit" />
      </Container>
      <List habits={habits} deleteHabit={deleteHabit} />
    </>
  )

  function handleSubmit(e) {
    e.preventDefault()
    const form = e.target
    const input = form.habit
    setHabits([input.value, ...habits])
    form.reset()
  }

  function deleteHabit(selectedHabit) {
    setHabits(habits.filter((habit) => selectedHabit !== habit))
  }
}

const Container = styled.form`
  margin-top: 20px;
  display: flex;
  justify-content: center;
`
const Input = styled.input`
  width: 300px;
  height: 30px;
`
