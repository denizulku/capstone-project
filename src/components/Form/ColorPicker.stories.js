import React from 'react'
import ColorPicker from './Form'
import { CirclePicker } from 'react-color'

export default {
  title: 'CirclePicker',
  component: ColorPicker,
}

const Template = () => <CirclePicker />

export const FormColorPicker = Template.bind({})
