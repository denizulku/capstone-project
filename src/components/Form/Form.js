import React from 'react'
import styled from 'styled-components'

export default function Form({ onSubmit }) {
  return (
    <>
      <FormContainer onSubmit={handleSubmit}>
        <InputStyled name="habit" type="text" placeholder="Insert a habit" />
      </FormContainer>
    </>
  )

  function handleSubmit(e) {
    e.preventDefault()
    const form = e.target
    const input = form.habit
    if (input.value === '') {
      alert('Please add a new habit')
    } else {
      onSubmit(input.value)
    }
    form.reset()
  }
}

const FormContainer = styled.form`
  margin-top: 20px;
  display: flex;
  justify-content: center;
`
const InputStyled = styled.input`
  width: 300px;
  height: 30px;
`
