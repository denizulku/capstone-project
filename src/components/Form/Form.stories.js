import React from 'react'
import Form from './Form'

export default {
  title: 'Form',
  component: Form,
}

const Template = (args) => <Form {...args} />

// Each story then reuses that template
export const Primary = Template.bind({})

Primary.args = {
  habits: ['foo', 'bar'],
}
Primary.argTypes = { onSubmit: { action: 'addHabit' } }
