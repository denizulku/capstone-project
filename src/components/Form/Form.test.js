import React from 'react'
import { render } from '@testing-library/react'
import Form from './Form.js'

test('the form data is correct', () => {
  expect(Form).toMatchSnapshot()
})
