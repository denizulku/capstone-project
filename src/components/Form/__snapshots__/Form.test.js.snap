// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the form data is correct 1`] = `[Function]`;
